src/Symfony/Component/Ldap/Adapter/ConnectionInterface.php

```diff
@@ -33,4 +34,24 @@ public function isBound(): bool;
      * @throws InvalidCredentialsException When the connection can't be created because of an LDAP_INVALID_CREDENTIALS error
      */
     public function bind(?string $dn = null, #[\SensitiveParameter] ?string $password = null): void;
+
+    /**
+     * Binds the connection against a user's DN and password using SASL
+     *
+     * @return void
+     *
+     * @throws LdapException               When SASL support is not available
+     * @throws AlreadyExistsException      When the connection can't be created because of an LDAP_ALREADY_EXISTS error
+     * @throws ConnectionTimeoutException  When the connection can't be created because of an LDAP_TIMEOUT error
+     * @throws InvalidCredentialsException When the connection can't be created because of an LDAP_INVALID_CREDENTIALS error
+     */
+    public function sasl_bind(?string $dn = null, #[\SensitiveParameter] ?string $password = null, ?string $mech = null, ?string $realm = null, ?string $authc_id = null, ?string $authz_id = null, ?string $props = null): void;
```

Comment:
please use camelCase everywhere in the PR: method names, arguments and variables